openml/extensions/__init__.py

```diff
@@ -0,0 +1,92 @@
+from typing import Any, Optional, Type, TYPE_CHECKING
+
+from .extension_interface import Extension
+
+if TYPE_CHECKING:
+    from openml.flows import OpenMLFlow
+
+
+extensions = []
+
+
+def register_extension(extension: Type[Extension]) -> None:
+    """Register an extension.
+
+    Registered extensions are considered by ``get_extension_by_flow`` and
+    ``get_extension_by_model``, which are used by ``openml.flow`` and ``openml.runs``.
+
+    Parameters
+    ----------
+    extension : Type[Extension]
+
+    Returns
+    -------
+    None
+    """
+    extensions.append(extension)
+
+
+def get_extension_by_flow(
+    flow: 'OpenMLFlow',
+    raise_if_no_extension: bool = False,
+) -> Optional[Extension]:
+    """Get an extension which can handle the given flow.
+
+    Iterates all registered extensions and checks whether they can handle the presented flow.
+    In case an extension can, it is immediately returned.
```

Comment:
I would word it slightly differently, from
`In case an extension can, it is immediately returned.`
to
`The first extension which can handle to flow is returned (even if multiple extensions can handle the flow).`
The part in parenthesis can be included or not based on what you feel is best.

Edit: I would probably leave out the word "all" from the first sentence too, since it does not necessarily iterate all extensions (as indicated by the second sentence).

Reply:
I totally agree and will add this.